src/utils/balancer-utils.ts

```
@@ -160,3 +160,21 @@ export const getBalancerToHlpPath = (from: TokenInfo, to: TokenInfo) => {
     tokenOut: path.tokenIn
   };
 };
+
+export const doesHlpToBalancerPathExist = (from: TokenInfo, to: TokenInfo) => {
+  try {
+    getHlpToBalancerPath(from, to);

```

Why would this throw an error? Why is that the way we determine if there's a path? What if there should be a path but the app threw an unrelated error?

---

The standard on handle that getter functions throw errors if they cannot get the thing, instead of returning undefined. If it fails for an unrelated reason, then the path should not be used in a route. Perhaps it makes sense to log the error?

---

That's insane. They should return undefined if they can't find something, not throw an error.

---

Also, a 'get' in the title looks like it should return something, not throw an error or otherwise to detemine if the thing can be got